librz/core/cil.c

```diff
@@ -574,35 +539,39 @@ RZ_IPI void rz_core_analysis_rzil_vm_status(RzCore *core, const char *var_name,
 		rzil_print_register_bitv(rzil->vm->pc, &p);
 	}
 
-	void **it;
-	rz_pvector_foreach (&rzil->vm->vm_global_variable_list, it) {
-		RzILVar *var = *it;
-		if (var_name && strcmp(var_name, var->var_name)) {
-			continue;
-		}
-		p.name = var->var_name;
-		RzILVal *val = rz_il_hash_find_val_by_var(rzil->vm, var);
-		switch (val->type) {
-		case RZ_IL_VAR_TYPE_BV:
-			rzil_print_register_bitv(val->data.bv, &p);
-			break;
-		case RZ_IL_VAR_TYPE_BOOL:
-			rzil_print_register_bool(val->data.b->b, &p);
-			break;
-		case RZ_IL_VAR_TYPE_UNK:
-			rzil_print_register_unk(&p);
-			break;
-		default:
-			rz_warn_if_reached();
-			break;
-		}
-		if (var_name) {
-			break;
-		}
-		if (rz_strbuf_length(p_sb(p.ptr)) > 95) {
-			rz_cons_printf("%s\n", rz_strbuf_get(p_sb(p.ptr)));
-			rz_strbuf_fini(p_sb(p.ptr));
+	RzPVector *global_vars = rz_il_vm_get_all_vars(rzil->vm, RZ_IL_VAR_KIND_GLOBAL);
```

Comment:
I think it worth extracting this vars initialization into a separate function. Could be reused somewhere else separately, as well.

Reply:
I'm not sure I understand what you mean. This is not initializing anything, it is only printing the var contents.